app/src/main/java/com/keylesspalace/tusky/TuskyApplication.kt

```diff
@@ -91,4 +99,18 @@ class TuskyApplication : Application(), HasAndroidInjector {
     }
 
     override fun androidInjector() = androidInjector
+
+    private fun upgradeSharedPreferences(oldVersion: Int, newVersion: Int) {
+        Log.d(TAG, "Upgrading shared preferences: $oldVersion -> $newVersion")
+        val editor = sharedPreferences.edit()
+
+        // Future upgrade code goes here
```

Comment:
How is this meant to be used?

There should be a number of steps - like in migrations?

Reply:
I have added some usage (fleshing out a loop) in my PR:

https://github.com/tuskyapp/Tusky/pull/3205/commits/cfe37d03808bbb6884112082680e108d374d7e1c

Reply:
Yes, that's a good approach.

Another hypothetical, in case it's useful.

Suppose we decided to change the default colour scheme from "dark" to "follow system", so that the user's first experience with Tusky matches their expressed preferences.

New users should get the new default, old users who didn't change the preference should get the old default. Otherwise old users who hadn't changed the default would see the colour scheme change.

So the migration path would check if the preference schema is being changed (changing the default value for a preference is considered a schema change), and would need to explicitly set an empty preference value to the old default value.

Ultimately all of this should be hidden behind a PreferenceRepository, but baby steps.